src/main/java/teammates/ui/controller/InstructorEditInstructorFeedbackPageAction.java

```diff
@@ -0,0 +1,105 @@
+
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.COURSE_ID), 
+                                 courseId);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_NAME), 
+                                 feedbackSessionName);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_MODERATED_STUDENT), 
+                                 moderatedEntityIdentifier);
+
+        new GateKeeper().verifyAccessible(
+                logic.getInstructorForGoogleId(courseId, account.googleId), 
+                logic.getFeedbackSession(feedbackSessionName, courseId),
+                false, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION);
+        
+        InstructorAttributes instructorUnderModeration = logic.getInstructorForEmail(courseId, moderatedEntityIdentifier);
+
+        // If the instructor doesn't exist
+        if (instructorUnderModeration == null) {
+            throw new EntityDoesNotExistException("Instructor Email "
+                    + moderatedEntityIdentifier + " does not exist in " + courseId
+                    + ".");
+        }
+
+        FeedbackSubmissionEditPageData data = new FeedbackSubmissionEditPageData(account, student);
+
+        data.bundle = logic.getFeedbackSessionQuestionsBundleForInstructor(
+                feedbackSessionName, 
+                courseId, 
+                instructorUnderModeration.email);
+
+        Assumption.assertNotNull(data.bundle);
+        
+        data.setSessionOpenForSubmission(true);
+        data.setModeration(true);
+        data.setHeaderHidden(true);
+        data.setPreviewInstructor(instructorUnderModeration);
+        data.setSubmitAction(Const.ActionURIs.INSTRUCTOR_EDIT_INSTRUCTOR_FEEDBACK_SAVE);
+
+        if (moderatedQuestionNumber != null) {
+            data.setModeratedQuestion(moderatedQuestionNumber);
+        }
+
+        hideQuestionsWithAnonymousResponses(data.bundle);
+
+        statusToAdmin = "Moderating feedback session for instructor (" + instructorUnderModeration.email + ")<br>" +
+                        "Session Name: " + feedbackSessionName + "<br>" +
+                        "Course ID: " + courseId;
+        
+        data.init("", "", courseId);
+        
+        return createShowPageResult(Const.ViewURIs.INSTRUCTOR_FEEDBACK_SUBMISSION_EDIT, data);
+    }
+    
+    /**
+     * Removes question from the bundle if the question has givers, recipients or responses that are anonymous to instructors.
+     * @param bundle
+     */
+    private boolean hideQuestionsWithAnonymousResponses(FeedbackSessionQuestionsBundle bundle) {
```

Comment:
This function seems to be duplicated from InstructorEditStudentFeedbackPageAction. Maybe we can find a better place for it so that there will not be repeated code.